packages/babel-register/src/node.js

```diff
@@ -7,6 +7,40 @@ import { OptionManager, DEFAULT_EXTENSIONS } from "@babel/core";
 import { addHook } from "pirates";
 import fs from "fs";
 import path from "path";
+import { Module } from "module";
+
+// Technically we could use the "semver" package here, but (for exmaple)
+// parseFloat("4.23.6") returns 4.23 so it's "good enough"
+const BABEL_SUPPORTS_EXTENSIONS_OPTION =
+  true || parseFloat(babel.version) >= 7.12;
```

Comment:
# :warning: 

- Before merging this PR, `7.12` needs to be replaced with the version this PR is released with.
- Before merging this PR, I need to add a warning in `new-version` in `Makefile` to remove `true`, which is only needed in tests before the actual release.